ibverbs/src/lib.rs

```diff
@@ -1304,6 +1301,51 @@ impl<T> MemoryRegion<T> {
             key: unsafe { &*self.mr }.rkey,
         }
     }
+
+    /// Make a subslice of this memory region.
+    pub fn slice(&mut self, bounds: impl RangeBounds<usize>) -> LocalMemorySlice<'_> {
```

Comment:
Because this takes `&mut self`, you'll still only be able to construct one such sub-slice at a time, which feels like it negates the intended outcome? In particular, you will _only_ be able to send/receive to multiple memory regions if they are sliced from _different_ `MemoryRegion` as currently written.

Reply:
I think what you'd have to do to enable multi-sub-slicing is the same thing `std` does for slices, which is to provide `split_at` and friends that return multiple guaranteed-non-overlapping sub-slices.

Reply:
Hmm, if these can be accessed after the fact, then `&` and `unsafe` isn't the fix either, since Rust still has a very fundamental assumption that things behind `&` will _never_ change while the `&` exists. This feeds into things like the optimizer, which is allowed to cache operations over values behind a `&` for example. The _only_ way it's legal in Rust to mutate through `&` is via `UnsafeCell` (or something that uses it internally).

It this suggests that we'll have to move to raw pointers (ie, `*mut`) for this API.